spring-cloud-starter-dataflow-metrics-collector/src/main/java/org/springframework/cloud/dataflow/metrics/collector/endpoint/MetricsCollectorEndpoint.java

```diff
@@ -45,61 +49,83 @@
+			}
+			if (streamMetrics != null) {
+				entries.add(streamMetrics);
 			}
-			entries.add(stream);
 		}
-		return new ResponseEntity<Collection<Stream>>(entries, HttpStatus.OK);
+
+		int totalPages = (entries.size() == 0) ? 0 : entries.size() / entries.size();
+		PagedResources.PageMetadata pageMetadata = new PagedResources.PageMetadata(entries.size(), 0, entries.size(),
+				totalPages);
+		PagedResources<StreamMetrics> pagedResources = new PagedResources<>(entries, pageMetadata,
+				ControllerLinkBuilder.linkTo(MetricsCollectorEndpoint.class).withRel(Link.REL_SELF));
+		return new ResponseEntity<PagedResources<StreamMetrics>>(pagedResources, HttpStatus.OK);
 	}
 
 	/**
-	 * Converts a denormalized view of each application instance metric ({@link ApplicationMetrics}) into a
-	 * hierarchical model {@link Stream}
+	 * Converts a denormalized view of each application instance metric
+	 * ({@link ApplicationMetrics}) into a hierarchical model {@link StreamMetrics}
 	 * @param applicationMetrics object to be converted
 	 * @param root The root object of the hierarchy - null if the first conversion
-	 * @return a hierarchical view of metrics using {@link Stream} as the root object
+	 * @return a hierarchical view of metrics using {@link StreamMetrics} as the root
+	 * object
 	 */
-	private Stream convert(ApplicationMetrics applicationMetrics, Stream root){
+	private StreamMetrics convert(ApplicationMetrics applicationMetrics, StreamMetrics root) {
 		Assert.notNull(applicationMetrics.getProperties().get(ApplicationMetrics.STREAM_NAME),
 				"Missing STREAM_NAME from metrics properties");
 		Assert.notNull(applicationMetrics.getProperties().get(ApplicationMetrics.APPLICATION_NAME),
 				"Missing APPLICATION_NAME from metrics properties");
 		Assert.notNull(applicationMetrics.getProperties().get(ApplicationMetrics.APPLICATION_GUID),
 				"Missing APPLICATION_GUID from metrics properties");
-		Stream stream = (root == null) ? new Stream((String) applicationMetrics.getProperties().get(ApplicationMetrics.STREAM_NAME)) : root;
+		StreamMetrics streamMetrics = (root == null)
+				? new StreamMetrics((String) applicationMetrics.getProperties().get(ApplicationMetrics.STREAM_NAME))
+				: root;
 		Application application = new Application(
 				(String) applicationMetrics.getProperties().get(ApplicationMetrics.APPLICATION_NAME));
 		Double incomeRate = YANUtils
-				.cast(findMetric(applicationMetrics.getMetrics(), MetricsAggregator.INPUT_METRIC_NAME).getValue(), Double.class)
+				.cast(findMetric(applicationMetrics.getMetrics(), MetricsAggregator.INPUT_METRIC_NAME).getValue(),
+						Double.class)
 				.orElse(0.0);
 		Double outgoingRate = YANUtils
-				.cast(findMetric(applicationMetrics.getMetrics(), MetricsAggregator.OUTPUT_METRIC_NAME).getValue(), Double.class)
+				.cast(findMetric(applicationMetrics.getMetrics(), MetricsAggregator.OUTPUT_METRIC_NAME).getValue(),
+						Double.class)
 				.orElse(0.0);
+
 		Integer instanceIndex = YANUtils
```

Comment:
Still getting 0 here. Might be that as actual json looks like
```
spring.application.index": "1",
```
so it might be that it's a `String`, not `Integer` so simple casting fails.